### PR TITLE
2nd Assignment by 202401193 (Kangjina)

### PR DESCRIPTION
**Pull request by kangjina4006-boop**
# Assignment Submission

## Student Information
- **Student ID**: [202401193]
- **Name**: [Kangjina]
- **Assignment**: [2]

## Changed Files

```
[FILE_TREE_PLACEHOLDER]
```

### Problem 1
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 2
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 3
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 4 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 5 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 6 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

## Testing
- [ ] I have run `python grader.py` locally
- [ ] All implemented problems pass their tests
- [ ] I have tested edge cases mentioned in the problem statements

## Additional Notes

---
**Note**: This PR will be automatically validated for:
- Correct directory structure
- Valid Python syntax
- PR title format: `Nth Assignment by {student_id} (Full Name)`
- File organization compliance

### CODE REVIEW

**Comment by kangjina4006-boop** [comment]
확인부탁드립니다!

File: submissions/202401193/assignment2/problem2/textops/clean/filters.py
Comment (hist0613) [inline submissions/202401193/assignment2/problem2/textops/clean/filters.py:1]: 구현이 비어있습니다~

**Comment by kangjina4006-boop** [comment]
수정했습니다! 그리고 assignment2 바로 밑에있는 README.md때문에 구조가 틀렸다고 뜨는 것 같은데 이걸 지워야 할까요..?

File: assignments/practice1/main.py
Comment (hist0613) [inline assignments/practice1/main.py:1]: `assignments/` 아래의 파일은 변경하시면 안 됩니다

File: submissions/202401193/assignment2/problem2/textops/tokenize/word.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    Split on single spaces; empty/whitespace-only -> [].
+    Assume `s` is already normalized by clean_text.
+    """
+    if s and s.strip():
+        li = s.split(" ")
+    else:
+        li = []
+    return li
+
+# 이 파트 제미나이의 도움을 받았습니다
```
Comment (hist0613) [inline submissions/202401193/assignment2/problem2/textops/tokenize/word.py:12]: good

File: submissions/202401193/assignment2/problem6/main.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        # 1) TP = 실제 양성을 양성으로 예측한 수
+        TP: int = sum(
+            1 for t, p in zip(y_true, y_pred)
+            if t == self.positive_class and p == self.positive_class
+        )
+        # 2) FP = 실제 음성을 양성으로 예측한 수
+        # 주의: 정밀도 공식은 TP / (TP + FP)이며, FP는 실제 **음성**인데 **양성**으로 예측한 경우입니다.
+        # "남의 코드"의 FP 주석은 사실과 다르게 보일 수 있으나 (t==positive_class가 아님)
+        # "남의 코드"의 FP 계산 로직은 정밀도 계산을 위해 TP + FP를 구하는 과정에서 오류가 있었습니다.
+        # 올바른 정밀도 분모 (TP + FP)를 위해, '양성으로 예측된 모든 경우'를 계산하고, 그중 실제 양성인 경우 (TP)를 나누어 계산합니다.
```
Comment (hist0613) [inline submissions/202401193/assignment2/problem6/main.py:95]: 이건 무슨 내용인가요?

**Comment by hist0613** [comment]
grader 기준 만점이지만, assignment 파일 변경 등 이슈가 있어서 approve는 아직 하지 않겠습니다
코멘트 달아놓은 부분들 확인해주세요

File: submissions/202401193/assignment2/problem6/main.py (hunk 2 above)
Comment (kangjina4006-boop) [reply submissions/202401193/assignment2/problem6/main.py:95]: 헉 제가 처음 쓴 코드가 오류 떠서 고쳐달라고 했는데 새로 달린 주석을 못 보고 그대로 올렸습니다.. 죄송합니다. 바로 수정하고 출처 남기겠습니다!
